simulator/tests/ExecutiveTests.ts

```diff
@@ -227,6 +228,35 @@ describe("Executive", () => {
         });
     });
 
+    it("sets the 'taskIndex' of the warrior to zero if the final task in the warrior has a DAT instruction executed", () => {
+
+        prepareCore(5, [
+            DataHelper.buildInstruction(3, OpcodeType.DAT, ModifierType.A, ModeType.Direct, 0, ModeType.Direct, 0)
+        ]);
+
+        const numberOfTasks = 3;
+        const expectedNumberOfTasks = numberOfTasks - 1;
+
+        var context = buildContext(3, 0, 0, 2, numberOfTasks);
+
+        var exec = new Executive(publisher);
+        exec.initialise(options);
+        exec.commandTable[OpcodeType.DAT].apply(exec, [context]);
+
+        expect(state.warriors[0].taskIndex).to.be.equal(0);
+        expect(state.warriors[0].tasks.length).to.be.equal(expectedNumberOfTasks);
+        expect(state.warriors[0].tasks[0].instructionPointer).to.be.equal(0);
+        expect(state.warriors[0].tasks[1].instructionPointer).to.be.equal(0);
+
+        expect(publisher.publish).to.be.calledWith({
```

Comment:
I think there is already a separate test for the publish event - this test has more than one reason to fail.

Reply:
That's true, but I didn't want to move too far from all the other tests in that file which seemed to use a similar approach. Happy to trim that bit though.